src/alloc-override.c

```diff
@@ -77,18 +77,12 @@ typedef void* mi_nothrow_t;
     MI_INTERPOSE_MI(calloc),
     MI_INTERPOSE_MI(realloc),
     MI_INTERPOSE_MI(strdup),
-    #if defined(MAC_OS_X_VERSION_10_7) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_7
```

Comment:
I don't think it's worth deleting it.
Instead replace all `MAC_OS_X_VERSION_MAX_ALLOWED` with `MAC_OS_X_VERSION_MIN_REQUIRED` in this file (or better yet, everywhere else in this repository).
FYI read thread in this pull request: https://github.com/hacl-star/hacl-star/pull/1042